### PR TITLE
Drop task on serialization error

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
- Drop task on unknown encoding and blob deserialization error, which is likely due to data corruption.
- Remove panic in NewDataBlob when encoding is invalid. Upon decoding, unknown encoding error will be returned.

**Why?**

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**
Yes

### CODE REVIEW

File: service/history/queues/executable.go [hunk 1]
```diff
@@ -516,3 +519,19 @@ func (e *executableImpl) estimateTaskMetricTag() []metrics.Tag {
 		metrics.OperationTag(taskType), // for backward compatibility
 	}
 }
+
+func (e *executableImpl) capturePanic(retError *error) {
+	if panicObj := recover(); panicObj != nil {
```
Comment (yycptt) [inline service/history/queues/executable.go:524]: This **_IS_** called inside a deferred function.

File: common/persistence/serialization/serializer.go [hunk 2]
```diff
@@ -107,17 +109,20 @@ type (
...
-		msg string
+		encodingType enumspb.EncodingType
+		wrappedErr   error
 	}
 
 	// UnknownEncodingTypeError is an error type for unknown or unsupported encoding type
 	UnknownEncodingTypeError struct {
-		encodingType enumspb.EncodingType
+		encodingTypeStr      string
+		supportedEncodingStr []string
```
[the remark this answers is not in the document]
Comment (yiminc) [reply common/persistence/serialization/serializer.go:125]: 👍

File: common/persistence/serialization/serializer.go [hunk 3]
```diff
@@ -254,30 +259,68 @@ func (t *serializerImpl) serialize(p proto.Marshaler, encodingType enumspb.Encod
 }
 
 // NewUnknownEncodingTypeError returns a new instance of encoding type error
-func NewUnknownEncodingTypeError(encodingType enumspb.EncodingType) error {
-	return &UnknownEncodingTypeError{encodingType: encodingType}
+func NewUnknownEncodingTypeError(
+	encodingTypeStr string,
+	supportedEncoding ...enumspb.EncodingType,
```
Comment (yiminc) [inline common/persistence/serialization/serializer.go:264]: Why do we need this? Isn't it always just the valid enum values of enumspb.EncodingType?

File: service/history/queues/executable.go (hunk 1 above)
Comment (yiminc) [reply service/history/queues/executable.go:524]: This require the capturePanic be called directly by a defer, like 
defer capturePanic(&retErr)
How do we prevent people do this:
defer func() { capturePanic(&retErr) }

File: common/persistence/serialization/serializer.go (hunk 3 above)
Comment (yycptt) [reply common/persistence/serialization/serializer.go:264]: In some cases, we assume the encoding is always proto3 I think. e.g. https://github.com/temporalio/temporal/pull/3803/files#diff-4354d109cf3924c1736b9d080b790089ec74eaca2fdd174624c2a8bf02bdd587R187 

I should probably rename this to `expectedEncoding`

File: service/history/queues/executable.go (hunk 1 above)
Comment (yycptt) [reply service/history/queues/executable.go:524]: I think now we have this linter that can help capture this issue? Or I can get rid of this function and embed it in Execute(), I don't think this function will/should be called in other places.

**Review by yiminc — APPROVED** [review]

**Pull request merged**